hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMCertStore.java

```diff
@@ -58,22 +61,43 @@ public void storeValidCertificate(BigInteger serialID,
     lock.lock();
     try {
       // This makes sure that no certificate IDs are reusable.
-      if ((getCertificateByID(serialID, CertType.VALID_CERTS) == null) &&
-          (getCertificateByID(serialID, CertType.REVOKED_CERTS) == null)) {
-        scmMetadataStore.getValidCertsTable().put(serialID, certificate);
-      } else {
-        throw new SCMSecurityException("Conflicting certificate ID");
-      }
+      checkValidCertID(serialID);
+      scmMetadataStore.getValidCertsTable().put(serialID, certificate);
     } finally {
       lock.unlock();
     }
   }
 
+  @Override
+  public void storeValidScmCertificate(BigInteger serialID,
```

Comment:
Can we add a storeValidScmCertificate method with a nodeType parameter instead of storeValidScmCertificate to avoid adding more interface changes in the future?

Reply:
Do you mean `storeValidCertificate`?

Reply:
updated.

Reply:
Done